src/platform/unix/mod.rs

```diff
@@ -1033,6 +1058,11 @@ fn is_socket(fd: c_int) -> bool {
 
 // FFI stuff follows:
 
+#[cfg(target_os="linux")]
+unsafe fn memfd_create(name: *const c_char, flags: usize) -> c_int {
+    syscall!(MEMFD_CREATE, name, flags) as c_int
```

Comment:
Kind of less than ideal in my books, but since glibc doesn't provide an wrapper, this is the recommended method. Perhaps we should make `shm_open` the default and this enabled by a feature flag?